### PR TITLE
feat(evm): namespace registration and querying precompile 

### PR DESCRIPTION
**Pull request by technicallyty**
## Overview

adds namespace registration and querying as a precompile to the EVM base shard.

### CODE REVIEW

File: evm/precompile/router/router_test.go
Comment (technicallyty) [inline evm/precompile/router/router_test.go:1]: removing the ginko tests from here cause they were copied from polaris and i also hate that cursed testing framework

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -22,7 +22,5 @@ jobs:
       #   uses: ScribeMD/docker-cache@0.3.4
       #   with:
       #     key: docker-${{ runner.os }}-${{ hashFiles('evm/Dockerfile') }}
-      - name: Forge Build evm/contracts
-        run: make forge-build
```
Comment (technicallyty) [inline .github/workflows/build.yml:26]: we don't actually need to do this, we have tests that cover if our contracts are out of date.

**Comment by technicallyty** [comment]
gonna close this, we're going to make namespace registration an automatic thing.

**Comment by namrapatel** [comment]
> gonna close this, we're going to make namespace registration an automatic thing.
family-man-man-gopher (1)

**Pull request closed without merging**